### ISSUE TITLE
Translations containing backslashes

### ISSUE THREAD

**Issue by patrickberkeley** [ember-intl/ember-intl#616]
- [x] I am on the [latest](https://github.com/ember-intl/ember-intl/releases/latest) ember-intl version
- [x] I have searched the [issues](https://github.com/ember-intl/ember-intl/issues) of this repo and believe that this is not a duplicate

### Environment

- Ember Version: 3.1.0
- Ember CLI Version: 3.1.0
- Ember Intl Version: 3.2.6
- Browser(s): Chrome latest
- Node Version: 8.9.4

### Steps to Reproduce

1. Add a translation that contains backslashes e.g.,

```yml
email:
    formats:
      default: '^.+@.+\.\w+$'
```

The reason we return a regex in translations it to ensure consistency in UI and API validations i.e., both the UI and API use this regex.

2. Try to load the translation in your app with `intl.t('email.formats.default')`

View the `peg$SyntaxError` error in the console: 

```
{
  "message": "Expected \"\\\\#\", \"\\\\\\\\\", \"\\\\u\", \"\\\\{\", \"\\\\}\", \"{\", [^{}\\\\\\0-\\x1F\\x7f \\t\\n\\r], end of input or whitespace but \"\\\\\" found.",
  "expected": [
    {
      "type": "literal",
      "value": "\\#",
      "description": "\"\\\\#\""
    },
    {
      "type": "literal",
      "value": "\\\\",
      "description": "\"\\\\\\\\\""
    },
    {
      "type": "literal",
      "value": "\\u",
      "description": "\"\\\\u\""
    },
    {
      "type": "literal",
      "value": "\\{",
      "description": "\"\\\\{\""
    },
    {
      "type": "literal",
      "value": "\\}",
      "description": "\"\\\\}\""
    },
    {
      "type": "literal",
      "value": "{",
      "description": "\"{\""
    },
    {
      "type": "class",
      "value": "[^{}\\\\\\0-\\x1F\\x7f \\t\\n\\r]",
      "description": "[^{}\\\\\\0-\\x1F\\x7f \\t\\n\\r]"
    },
    {
      "type": "end",
      "description": "end of input"
    },
    {
      "type": "other",
      "description": "whitespace"
    }
  ],
  "found": "\\",
  "location": {
    "start": {
      "offset": 6,
      "line": 1,
      "column": 7
    },
    "end": {
      "offset": 7,
      "line": 1,
      "column": 8
    }
  },
  "name": "SyntaxError"
}
```

**Comment by jasonmit** [comment]
Upstream issue in [yahoo/intl-messageformat-parser](https://github.com/yahoo/intl-messageformat-parser/).  

Also related: https://github.com/yahoo/intl-messageformat/issues/109 https://github.com/yahoo/react-intl/issues/1182

I'll keep this issue open for tracking.

**Comment by patrickberkeley** [comment]
Thanks for the info, @jasonmit. Would you like me to close the issue here? Or how would you like to proceed?

**Comment by jasonmit** [comment]
We should keep this open for tracking and investigate removing backslash as escalpe character in favor of the quote escaping that the spec mentions.

**Comment by sandstrom** [comment]
> 've begun to port the logic over to the intl-messageformat-parser.

Is this about moving to use the node module `intl-messageformat-parser` directly? (https://github.com/formatjs/formatjs/tree/master/packages/intl-messageformat-parser)

**Comment by jasonmit** [comment]
@sandstrom work was put on pause.  Was talking to @kturney about possibly replacing the parser/compiler which is mentioned here: https://github.com/ember-intl/ember-intl/issues/811#issuecomment-468790810

The migration fixes a number of issues and unlocks custom formatters.  May be a steep cost to migrate and if so, we can likely tackle getting us back on HEAD of the intl-messageformat-parser and compiler with more ease in the meantime.

That said, I don't have the time to dedicate towards this effort in the short term.

**Comment by sandstrom** [comment]
Thanks for the update @jasonmit! I ran across that fork of intl-message-format-parser when I tried to track down a syntax error we had. I've opened a separate PR for that (https://github.com/ember-intl/ember-intl/pull/1179).

I agree the change may be useful, since we'd offload/share some of the work with the [other repo](https://github.com/formatjs/formatjs/tree/master/packages/intl-messageformat-parser).

**Comment by jasonmit** [comment]
After updating the translation parser and compiler, this should now be resolved on master.  Single quotes now properly escapes characters within the ICU syntax.

There are many breaking changes so will be releasing a 5.0.0-unstable shortly while I document all the breaking changes.

**Comment by jasonmit** [comment]
This should now be resolved in 5.0.0, single quotes are now the method of escaping (as per the ICU spec).
